drivers/entropy/entropy_sam.c

```diff
@@ -18,7 +18,7 @@ struct trng_sam_dev_cfg {
 #define DEV_CFG(dev) \
 	((const struct trng_sam_dev_cfg *const)(dev)->config->config_info)
 
-static int entropy_sam_wait_ready(Trng *const trng)
+static int entropy_sam_wait_ready(Trng * const trng, u32_t flags)
 {
```

Comment:
I think the problem with your implementation/changes is that the original intent of entropy_get_entropy_isr() when the blocking flag is zero is to just get as much entropy data is available then return where the return value is the number of bytes retrieved. Note that this may not be equal to the number of bytes requested so it is up to the caller to request the rest.